util/otbn_build.py

```diff
@@ -170,17 +170,49 @@ def main() -> int:
         out_elf = out_dir / (app_name + '.elf')
         call_otbn_ld(obj_files, out_elf, linker_script = args.linker_script)
 
+        # Use objcopy to create an ELF that can be linked into a RISC-V binary
+        # (to run on Ibex). This should set flags for all sections like rodata
+        # (since they're not executable on Ibex, nor does it make sense for
+        # Ibex code to manipulate OTBN data sections "in place"). We name them
+        # with a .otbn prefix, so end up with e.g. .rodata.otbn.text and
+        # .rodata.otbn.data.
+        #
+        # There are 4 symbols that we want to keep and link in, giving
+        # the start and end of imem and dmem. These will be relocated as part
+        # of the link, so they'll give addresses in the Ibex address space. So
+        # that the RISC-V binary can link multiple OTBN applications, we give
+        # them an application-specific prefix. (Note: This prefix is used in
+        # sw/device/lib/runtime/otbn.h: so needs to be kept in sync with that).
+        #
+        # The other symbols in the OTBN binary are not of interest to the
+        # RISC-V code, and they occupy a different namespace (referring to
+        # addresses in OTBN memories, rather than addresses in the rodata
+        # sections). Strip them out.
```

Comment:
I am not familiar how the loading and execution process works for OTBN apps, but my impression was that for calling a function not only the memory region symbols are needed but also the offset of functions. For example in `dif_otbn_smoketest.c`
```c
OTBN_PTR_T_INIT(barrett384, wrap_barrett384);
otbn_call_function(otbn_ctx, kFuncWrapBarrett384)
```
And for this the symbol of the function must also be present.

Reply:
Oh, good point (and the code didn't fail CI because I'd forgotten the `--strip-all` parameter). I'll get rid of this stripping entirely.